src/vm/dllimport.cpp

```diff
@@ -6118,6 +6104,36 @@ static void DetermineLibNameVariations(const char* const** libNameVariations, in
         *numberOfVariations = COUNTOF(NameOnly);
     }
 }
+#else
+static void DetermineLibNameVariations(const char* const** libNameVariations, int* numberOfVariations, const SString& libName, bool libNameIsRelativePath)
+{
+    bool endWithDot = libName.EndsWith(W("."));
+    bool notContainDot = !libName.Find(libName.Begin(), W('.'));
+    bool containsSuffix = libName.EndsWithCaseInsensitive(W(".dll")) || libName.EndsWithCaseInsensitive(W(".exe"));
+    if (endWithDot || notContainDot || containsSuffix || !libNameIsRelativePath)
+    {
+        // file name end with '.'
```

Comment:
This comment is not very useful. It is a literal rewrite of the above condition in plain English.

Instead, it would be useful to have comment that describes the reasons for this logic.

Reply:
Added more comments